testsuite/tests/mgc/dns/test_delete_gw.py

```diff
@@ -0,0 +1,34 @@
+"""Tests that DNSPolicy reacts correctly, when its underlying Gateway is deleted"""
+from time import sleep
+
+import pytest
+
+from testsuite.openshift.objects.dnspolicy import DNSPolicy
+
+pytestmark = [pytest.mark.mgc]
+
+
+def test_delete_gw(upstream_gateway, dns_policy, route):
+    """Tests that DNSPolicy reacts correctly, when its underlying Gateway is deleted"""
+
+    def delete_gw_error(policy: DNSPolicy):
+        for condition in policy.model.status.conditions:
+            if condition.type == "Ready" and condition.status == "False" and condition.reason == "TargetNotFound":
+                return True
+        return False
+
+    # test that it works before the policy
+    backend_client = route.client(verify=False)
+    sleep(30)  # DNS
+
+    response = backend_client.get("get")
+    assert response.status_code == 200, "Original DNSPolicy does not work"
+
+    upstream_gateway.delete()
+
+    # Wait for expected status
+    assert dns_policy.wait_until(delete_gw_error), "Policy did not reach expected status"
+
+    # It should no longer work
+    response = backend_client.get("get")
+    assert response.status_code == 404
```

Comment:
this is not a correct behavior since 404 means that there was somebody who responded 404 but there shouldn't be anybody listening on the IP; even hostname shouldn't be resolvable